tests/integration/test_dbapi_integration.py

```diff
@@ -615,13 +615,22 @@ def test_boolean_query_param(trino_connection):
     assert rows[0][0] is False
 
 
-def test_float_query_param(trino_connection):
-    cur = trino_connection.cursor()
-    cur.execute("SELECT ?", params=(1.1,))
+def test_row(trino_connection):
+    cur = trino_connection.cursor(experimental_python_types=True)
+    params = (1, Decimal("2.0"), datetime(2020, 1, 1, 0, 0, 0))
+    cur.execute("SELECT ?", (params,))
     rows = cur.fetchall()
 
-    assert cur.description[0][1] == "double"
-    assert rows[0][0] == 1.1
+    assert rows[0][0] == params
+
+
+def test_nested_row(trino_connection):
+    cur = trino_connection.cursor(experimental_python_types=True)
+    params = ((1, "test", Decimal("3.1")), Decimal("2.0"), datetime(2020, 1, 1, 0, 0, 0))
+    cur.execute("SELECT ?", (params,))
+    rows = cur.fetchall()
+
+    assert rows[0][0] == params
```

Comment:
Let's add another test case having field names:
```sql
SELECT CAST(ROW(1, 2e0) AS ROW(x BIGINT, y DOUBLE))
```

Reply:
I tested that yesterday, it has no impact on the data returned, if we want aliases of columns or fields we have to implement a DictCursor. Note that I also chose to use tuples instead of dicts for representing rows.

Not sure if we should revisit that decision to use tuples? From a Pyton API point of view It definitely feels more natural to pass down tuples as unnamed rows as query params compared to integer indexed dicts. WDYT?

Reply:
I agree for anonymous rows tuples make more sense.